dcos_launch/gce.py

```diff
@@ -0,0 +1,81 @@
+import json
+import logging
+
+from dcos_launch import util
+from dcos_test_utils import gce
+from dcos_test_utils.helpers import Host
+
+from googleapiclient.errors import HttpError
+
+log = logging.getLogger(__name__)
+
+
+class BareClusterLauncher(util.AbstractLauncher):
+    # Launches a homogeneous cluster of plain GMIs intended for onprem DC/OS
+    def __init__(self, config: dict):
+        credentials_path = util.set_from_env('GOOGLE_APPLICATION_CREDENTIALS')
+        credentials = util.read_file(credentials_path)
+        self.gce_wrapper = gce.GceWrapper(json.loads(credentials), credentials_path)
+        self.config = config
+
+    @property
+    def deployment(self):
+        """ Builds a BareClusterDeployment instance with self.config, but only returns it successfully if the
+        corresponding real deployment (active machines) exists and doesn't contain any errors.
+        """
+        try:
+            deployment = gce.BareClusterDeployment(self.gce_wrapper, self.config['deployment_name'],
+                                                   self.config['gce_zone'])
+            info = deployment.get_info()
+            errors = info['operation'].get('error')
+            if errors:
+                raise util.LauncherError('DeploymentContainsErrors', str(errors))
+            return deployment
+        except HttpError as e:
+            if e.resp.status == 404:
+                raise util.LauncherError('DeploymentNotFound', "The deployment you are trying to access doesn't exist")\
```

Comment:
Prefer breaking the line at the comma instead of using a backslash.
- https://google.github.io/styleguide/pyguide.html#Line_length
- note: we extend the line length requirement locally to 120 characters